facebook_connector.py

```diff
@@ -45,8 +53,23 @@ def resolve_user_id(user_id):
     Returns:
         The name of the user if it is able to find it, otherwise the input
     """
-    # TODO (neitsch): This method is not implemented yet.  Use the user_id
-    # (format: 12345678@facebook.com) to get the actual name of the user
+    fb_id_pattern_match = _FB_ID_PATTERN.match(user_id)
+    if fb_id_pattern_match:
+        fb_numeric_id = fb_id_pattern_match.group(1)
+        if fb_numeric_id not in _mapped_fb_ids:
+            username = fb_numeric_id
+            try:
+                fb_user_page = urlopen("https://www.facebook.com/{}".format(fb_numeric_id))
```

Comment:
This seems to fail for me quite a bit. That's fine if it's best effort, just clarify the documentation/print statements.

Also, do you want to have a boolean passed into `get_cleaned_fully_merged_messages()` to trigger calling this?